src/Voron/Impl/Scratch/PageFromScratchBuffer.cs

```diff
@@ -14,52 +16,63 @@ public bool Equals(PageFromScratchBuffer x, PageFromScratchBuffer y)
             if (x == y) return true;
             if (x == null || y == null) return false;            
 
-            return x.PositionInScratchBuffer == y.PositionInScratchBuffer && x.Size == y.Size && x.NumberOfPages == y.NumberOfPages && x.ScratchFileNumber == y.ScratchFileNumber;
+            return x.PositionInScratchBuffer == y.PositionInScratchBuffer && x.NumberOfPages == y.NumberOfPages && x.File == y.File;
```

Comment:
```
x.File == y.File
```

is this correct? this is ref comparison?

Reply:
Yes, it is meant to be a ref compare